packages/cli/src/tools/installPods.ts

```diff
@@ -15,7 +15,44 @@ type PromptCocoaPodsInstallation = {
   promptQuestion: string;
 };
 
-async function updatePods(loader: ora.Ora) {
+async function runPodInstall(
+  loader: ora.Ora,
+  projectName: string,
+  shouldHandleRepoUpdate: boolean = true,
+) {
+  try {
+    loader.start(
+      `Installing CocoaPods dependencies ${chalk.dim(
+        '(this may take a few minutes)',
+      )}`,
+    );
+    await execa('pod', ['install']);
+  } catch (error) {
+    // "pod" command outputs errors to stdout (at least some of them)
+    const stderr = error.stderr || error.stdout;
+
+    /**
+     * If CocoaPods failed due to repo being out of date, it will
+     * include the update command in the error message.
+     *
+     * `shouldHandleRepoUpdate` will be set to `false` to
+     * prevent infinite loop (unlikely scenario)
+     */
+    if (stderr.includes('pod repo update') && shouldHandleRepoUpdate) {
+      await runPodUpdate(loader);
+      await runPodInstall(loader, projectName, false);
+    } else {
+      loader.fail();
+      throw new Error(
```

Comment:
Let's use `CLIError` and pass the original `error` so that [issues like this one](https://github.com/react-native-community/cli/issues/528) are easier to deal with. Would be cool to do the same for `runPodUpdate` function (line 68)

Reply:
Was just copying and pasting the existing code and actually started thinking why we no longer use CLIError.

Reply:
Turns out this will take more than should be in scope of this PR and I guess we should track it separately.

CLIError will inline the entire message into a single line + we already `logger.error` here (in other CLI commands, we don't do that as we have global try-catch). 

I would be a bit nervous about changing it here due to potential side effects. What you think of setting up a follow up issue?

Reply:
I'm good with a followup